uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/components/leave-request-popup-details-tab.component.spec.js

```diff
@@ -1653,20 +1650,40 @@ define([
           delete toilRequest.toil_expiry_date;
 
           AbsenceType.canExpire.and.returnValue($q.resolve(true));
-          compileComponent({
-            mode: 'edit',
-            leaveType: 'toil',
-            request: toilRequest
+        });
+
+        describe('when request date changes for manager', function () {
+          beforeEach(function () {
+            compileComponent({
+              mode: 'edit',
+              leaveType: 'toil',
+              role: 'manager',
+              request: toilRequest
+            });
+            controller.request.to_date = new Date();
+            controller.onDateChangeExtended();
+            $rootScope.$digest();
+          });
+
+          it('update the toil expiry date', function () {
+            expect(controller.request.toil_expiry_date).toBeDefined();
           });
         });
 
-        describe('when request date changes', function () {
+        describe('when request date changes for staff', function () {
           beforeEach(function () {
+            compileComponent({
+              mode: 'edit',
+              leaveType: 'toil',
+              role: 'staff',
+              request: toilRequest
+            });
             controller.request.to_date = new Date();
+            controller.onDateChangeExtended();
             $rootScope.$digest();
           });
 
-          it('does not update the toil expiry date', function () {
+          it('update the toil expiry date', function () {
```

Comment:
Description says `update the toil expiry date` but the code is checking `toBeUndefined();`. Are you sure this is correct. Also its better to refactor it like
```
var valueBeforeDateChanged = <store value>;

expect(controller.request.toil_expiry_date).not.toBe(valueBeforeDateChanged);
```